### ISSUE TITLE
Support input-channel MX quantization for nn.Conv2d weights

### ISSUE THREAD

**Issue by catcor01** [pytorch/ao#4866]
## Description

`MXDynamicActivationMXWeightConfig` currently uses `MXTensor.to_mx()`, which forms blocks along the final tensor dimension.

PyTorch Conv2d weights use OIHW layout:

```text
[output_channels, input_channels/groups, kernel_height, kernel_width]
```

MX Conv2d requires blocks along `input_channels/groups`, which is dimension 1 rather than the final dimension.

The desired logical representation is:

```text
qdata:  [O, I/groups, KH, KW]
scales: [O, (I/groups)/block_size, KH, KW]
```

### Proposed adapter

We propose adding an internal Conv2d weight adapter to the registered `MXDynamicActivationMXWeightConfig` transformation.

The adapter temporarily moves the logical input-channel dimension to the final dimension:

```text
OIHW
  → OHWI
  → MXTensor.to_mx()
  → restore OIHW qdata
  → restore compact OIHW scale coordinates
```

This is a one-time model-conversion transformation. It does not add runtime permutations or change the public `MXTensor.to_mx()` API.

An illustrative implementation is:

```python
def quantize_conv2d_weight_to_mx(
    weight_oihw: torch.Tensor,
    config: MXDynamicActivationMXWeightConfig,
    act_quant_kwargs,
) -> MXTensor:
    # Move the logical input-channel dimension to the dimension
    # currently supported by MXTensor.to_mx().
    weight_ohwi = weight_oihw.permute(0, 2, 3, 1).contiguous()

    weight_mx_ohwi = MXTensor.to_mx(
        weight_ohwi,
        elem_dtype=config.weight_dtype,
        block_size=config.block_size,
        scaling_mode=config.scaling_mode,
        kernel_preference=config.kernel_preference,
        is_swizzled_scales=False,
    )

    # Restore PyTorch Conv2d logical coordinates.
    qdata_oihw = weight_mx_ohwi.qdata.permute(0, 3, 1, 2).contiguous()
    scale_oihw = weight_mx_ohwi.scale.permute(0, 3, 1, 2).contiguous()

    return MXTensor.from_qdata_and_scales(
        qdata_oihw,
        scale_oihw,
        orig_dtype=weight_oihw.dtype,
        block_size=config.block_size,
        kernel_preference=config.kernel_preference,
        act_quant_kwargs=act_quant_kwargs,
        is_swizzled_scales=False,
    )
```

The registered transformation would select this adapter for Conv2d:

```python
@register_quantize_module_handler(MXDynamicActivationMXWeightConfig)
def transform(module, config, *, parameter_name="weight"):
    if isinstance(module, torch.nn.Conv2d):
        quantized_weight = quantize_conv2d_weight_to_mx(
            getattr(module, parameter_name),
            config,
            create_activation_quantization_kwargs(config),
        )

        setattr(
            module,
            parameter_name,
            torch.nn.Parameter(quantized_weight, requires_grad=False),
        )
        return module

    # Preserve the existing Linear transformation.
    ...
```

The exact helper names and integration structure are illustrative.

### Public workflow

Users continue to use the existing TorchAO API and standard filters:

```python
def is_conv2d(module, fqn):
    return isinstance(module, torch.nn.Conv2d)


quantize_(
    model,
    MXDynamicActivationMXWeightConfig(
        weight_dtype=torch.float8_e4m3fn,
        activation_dtype=torch.float8_e4m3fn,
        block_size=32,
    ),
    filter_fn=is_conv2d,
)
```

### Expected behaviour

- Preserve `nn.Conv2d` and its attributes.
- Quantize weights along the logical input-channel dimension.
- Preserve OIHW qdata and compact scale coordinates.
- Avoid exposing backend packing or scale swizzling.
- Define behaviour when the input-channel dimension is not divisible by the block size.
- Support or clearly reject grouped and depthwise Conv2d.
- Leave the existing Linear transformation unchanged.
- Target MXFP8 initially. Other MXFP formats, including MXFP4 and MXFP6, can be addressed in follow-up work.

A general axis-aware `MXTensor.to_mx()` could replace the internal adapter later, but it is not required for this initial change.

This request concerns constant-weight conversion and representation. It does not require TorchAO to provide an optimized MX Conv2d kernel or perform runtime Conv2d activation quantization.

## Related work

- MX casting tracker: #1768
- Equivalent Conv2d structural-axis problem: #4626

## Target hardware

All / Not hardware specific

**Comment by abhinav-phi** [comment]
Hi! I'd like to work on this.

Plan: I'll design the feature against the described behavior, implement it with the project's conventions, add unit/integration tests, and update the docs.

Could you please assign this issue to me? Thanks!

/assign take
